org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterIlluminance.java

```diff
@@ -13,154 +13,47 @@
-                handleReportingResponse(reportingResponse, configReporting.getPollingPeriod(),
-                        configReporting.getReportingTimeMax());
-            } catch (InterruptedException | ExecutionException e) {
-                logger.debug("{}: Illuminance measurement exception setting reporting", endpoint.getIeeeAddress(), e);
-            }
-        }
-    }
-
-    @Override
-    public Channel getChannel(ThingUID thingUID, ZigBeeEndpoint endpoint) {
-        if (endpoint.getInputCluster(ZclIlluminanceMeasurementCluster.CLUSTER_ID) == null) {
-            logger.trace("{}: Illuminance measurement cluster not found", endpoint.getIeeeAddress());
-            return null;
-        }
-        return ChannelBuilder
-                .create(createChannelUID(thingUID, endpoint, ZigBeeBindingConstants.CHANNEL_NAME_ILLUMINANCE_VALUE),
-                        ZigBeeBindingConstants.ITEM_TYPE_NUMBER)
-                .withType(ZigBeeBindingConstants.CHANNEL_ILLUMINANCE_VALUE)
-                .withLabel(ZigBeeBindingConstants.CHANNEL_LABEL_ILLUMINANCE_VALUE)
-                .withProperties(createProperties(endpoint)).build();
-    }
-
-    @Override
-    public void attributeUpdated(ZclAttribute attribute, Object val) {
-        logger.debug("{}: ZigBee attribute reports {}", endpoint.getIeeeAddress(), attribute);
-        if (attribute.getCluster() == ZclClusterType.ILLUMINANCE_MEASUREMENT
-                && attribute.getId() == ZclIlluminanceMeasurementCluster.ATTR_MEASUREDVALUE) {
-            updateChannelState(new DecimalType(Math.pow(10.0, (Integer) val / 10000.0) - 1));
-        }
+public class ZigBeeConverterIlluminance extends ZigBeeReportableAttributeConverter {
+
+    private static final boolean IS_ANALOGUE = true;
+    private static final String NAME_FOR_LOGGING = "Illuminance measurement";
+    private static final Logger logger = LoggerFactory.getLogger("illuminasty");
+    private static final int CLUSTER_ID = ZclIlluminanceMeasurementCluster.CLUSTER_ID;
+    private static final int ATTRIBUTE_ID = ATTR_MEASUREDVALUE;
+    private static final String CHANNEL_NAME = CHANNEL_NAME_ILLUMINANCE_VALUE;
+    private static final String LABEL = CHANNEL_LABEL_ILLUMINANCE_VALUE;
+    private static final String ITEM_TYPE = ITEM_TYPE_NUMBER;
+    private static final ChannelTypeUID CHANNEL_TYPE_UID = CHANNEL_ILLUMINANCE_VALUE;
+    protected static BigDecimal CHANGE_DEFAULT = new BigDecimal(5000);
+    protected static BigDecimal CHANGE_MIN = new BigDecimal(10);
+    protected static BigDecimal CHANGE_MAX = new BigDecimal(20000);
+
+    public ZigBeeConverterIlluminance() {
+        super(ATTRIBUTE_ID, CLUSTER_ID, CHANNEL_NAME, LABEL, ITEM_TYPE, CHANNEL_TYPE_UID, IS_ANALOGUE, CHANGE_DEFAULT,
+                CHANGE_MIN, CHANGE_MAX, logger, NAME_FOR_LOGGING);
+    }
+
+    @Override
+    public State convertValueToState(Integer val) {
+        logger.debug("converting illuminace from {}", val);
```

Comment:
Probably this can be removed since there is already logging of the value in the attribute listener so this adds no further information.